src/Dofs/ConstraintHandler.jl

```diff
@@ -1059,7 +1059,7 @@ function _add!(ch::ConstraintHandler, pdbc::PeriodicDirichlet, interpolation::In
                field_dim::Int, offset::Int, is_legacy::Bool, rotation_matrix::Union{Matrix{T},Nothing}, ::Type{dof_map_t}, iterator_f::F) where {T, dof_map_t, F <: Function}
     grid = ch.dh.grid
     face_map = pdbc.face_map
-    Tx = typeof(first(ch.dh.grid.nodes).x) # Vec{D,T}
+    Tx = typeof(getcoordinates(first(getnodes(ch.dh.grid)))) # Vec{D,T}
```

Comment:
```suggestion
    Tx = get_coordinate_eltype(ch.dh.grid) # Vec{D,T}
```
?

Reply:
That returns the element type of the coordinate